### ISSUE TITLE
Adding RRest

### ISSUE THREAD

**Issue by avivrosenberg** [physiozoo/physiozoo#7]
Hi, @peterhcharlton, welcome!

Some issues/questions to address in order to incorporate the RRest code.

1. Naming. We have an unfortunate naming collision as "RR" is both for “respiratory rate” and also by much of the HRV code to mean R-R intervals. To make it clearer for users, I suggest to put the algorithmic code into two [Matlab packages](https://www.mathworks.com/help/matlab/matlab_oop/scoping-classes-with-packages.html), one for HRV analysis and one for respiratory rate analysis. Then users can call the functions with e.g. `mhrv.ecgrr(…)` or `resp.rrest(…)`. See below.

1. Repo organization. How would you like to add your code? Do you want to maintain a separate copy of part of the RRest code within the PZ repo? This is the easiest, but will require you to maintain changes in two repos (your main repo and the PZ repo). For the `mhrv` library, I have a separate [dedicated repo](https://github.com/physiozoo/mhrv) and then I use a feature of git (subtrees) to sync commits from that repo to the PZ repo (this one).  It’s possible to start simple (just put some of your code in the PZ repo, maintain it manually) and later on consolidate everything. So, do you want me to just add the .mat files you sent us to this repo?

1. Documentation. There are some technical issues I need to address to get the documentation generator (Sphinx) to work with both packages. I’ll fix it. In any case, I can create some template files for you to fill in with documentation about your package. The documentation for the functions will be loaded from the code itself, but any other docs require you to manually create them (e.g. tutorials or getting started pages and so on). See for example the sections in the [mhrv docs folder](https://github.com/physiozoo/physiozoo/tree/master/mhrv/docs/sections). Everything should then be generated together and hosted on the [PZ docs site](https://docs.physiozoo.com/en/latest/).

To explain the above, the project structure should be something like this:
```
PZ/
	...
	GUI/	(existing UI code)
	lib/ 	(new top-level folder for libraries used by PZ)
		mhrv/
			docs/
			+mhrv/ 	(this is Matlab package containing HRV code)
				mhrv.m
				mhrv_batch.m
				...
			... 
		resp/
			docs/	(your docs will be here)
			+resp/	(this is Matlab package containing your code)
				rrest.m
				setup_universal_params.m
				...
	...
```

I would appreciate any feedback or suggestions.

**Comment by avivrosenberg** [comment]
@peterhcharlton any feedback on this?

**Comment by peterhcharlton** [comment]
Hi,

My apologies for a lack of contact from me, unfortunately I haven't had the
time to dedicate to this. I hope to be able to give it some time in early
April. To briefly answer your questions:

1. Yes, happy to use an alternative name for Respiratory Rate to avoid the
similarity between names.

2. I think this is a key question, and I'm not yet sure of the answer. I am
reluctant to maintain two "RRest" repositories (one the original one, and
one the Physiozoo one), but at the moment I can't see another way.

3. I think at the moment it would be best for me to focus on the code
itself. There is already a published tutorial on how to use the code (here:
https://doi.org/10.1007/978-3-319-43742-2_26 ), and getting started pages
here: https://github.com/peterhcharlton/RRest/wiki/Getting-Started .
Consequently, I think the most pressing to-do for me is to load the code
onto PhysioZoo, rather than create new tutorials. I will of course be happy
to consider adding to this documentation once the code is ready.

I also have the remaining to-do of providing some sample data to use with
the code. I have been thinking about how best to do this, and expect to use
the datasets available here:
http://peterhcharlton.github.io/RRest/datasets.html .

Do let me know how this sounds to you.

Many thanks, and my apologies once again.

Pete

**Comment by avivrosenberg** [comment]
@peterhcharlton Thanks for your feedback.

Ok, so I will begin by restructuring the project into two packages and incorporate your code.
Later on, after that works and you can maintain it without issue, we'll focus on adding documentation for RRest you our documentation generator.

**Comment by peterhcharlton** [comment]
Hi,

My apologies, I haven't understood what you meant by "restructuring the project" - would you mind passing on further details? I am hoping to make time this week to get the code running with some example data, as discussed previously.

Best wishes,

Pete

**Comment by avivrosenberg** [comment]
I meant split the physiozoo code into two matlab packages, one for hrv and for respiration.
This will prevent name collisions an confusion about what RR is. It also means that users will need to call the code with the package name, e.g. `mhrv.ecgrr(...)` instead of just `ecgrr(...)`.
(see also example in my first message in this thread).

I haven't yet got around to doing this ...

**Comment by peterhcharlton** [comment]
That sounds like a good idea. It will also make it easier to add further packages in the future.

I'll try to update you on my progress over the coming days

Thanks,

Pete

**Comment by avivrosenberg** [comment]
Hi @peterhcharlton,
I'm done restructuring and project.
Sorry for the delay. I also had to fix the documentation generator to work with the new package-based structure so it took a bit longer than I thought as there were some technical issues.

The new structure is in the `package` branch and i created a new PR (#8) to track the changes and progress.

You can now look at that branch and see what you need to do. Basically:
- add your code under `lib/resp`.
- add a new section in the documentation (just copy what `mhrv` is doing there). I will gladly help if you have questions about this.

You can add your changes to the `package` branch and we can discuss them on the PR page (#8).

Thanks again.
